include/zephyr/cpu_freq/pstate.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2025 Analog Devices, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_SUBSYS_CPU_FREQ_PSTATE_H_
+#define ZEPHYR_SUBSYS_CPU_FREQ_PSTATE_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <zephyr/types.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/cpu_freq/pstate.h>
+
+/** Synthesize symbol of pstate from devicetree dependency ordinal */
+#define PSTATE_DT_SYM(_node) _CONCAT(__pstate_, DT_DEP_ORD(_node))
+
+/**
+ * @brief Define all P-state information for the given node identifier.
+ *
+ * @param _node Node identifier.
+ * @param _config Pointer to the SoC specific configuration for the P-state.
+ */
+#define PSTATE_DT_DEFINE(_node, _config)                                                           \
+	const struct pstate PSTATE_DT_SYM(_node) = {                                               \
+		.load_threshold = DT_PROP(_node, load_threshold),                                  \
+		.disabled = false,                                                                 \
```

Comment:
Given this struct is const, this member is redundant (its basically a #define 0)